src/build_vehicle_gui.cpp

```diff
@@ -1665,6 +1670,12 @@ struct BuildVehicleWindow : Window {
 		return list;
 	}
 
+	DropDownList BuildBadgeConfigurationList() const
+	{
+		static const auto separators = { STR_BADGE_CONFIG_PREVIEW, STR_BADGE_CONFIG_NAME };
+		return ::BuildBadgeClassConfigurationList(static_cast<GrfSpecFeature>(GSF_TRAINS + this->vehicle_type), 3, separators);
```

Comment:
Please use a named constant for the number of columns, as I'm seeing this 3 being used later in the class. Similarly there's a 2 that I suspect to be 'number of columns - 1'.